MonkeyLoader.Resonite.Integration/UI/Inspectors/CustomInspectorInjector.cs

```diff
@@ -17,61 +19,136 @@ internal sealed class CustomInspectorInjector
 
-            if (worker is ICustomInspector customInspector)
+            CodeInstruction[] instArr = instructions.ToArray();
+            for (int i = 0; i < instArr.Length; i++)
             {
-                try
+                var instruction = instArr[i];
+                bool branchFound = false;
+                if (afterHeaderBranchLabel is null && instruction.opcode == OpCodes.Brtrue && instArr[i-1].opcode == OpCodes.Isinst && instArr[i - 1].operand == (object)typeof(Slot))
                 {
-                    ui.Style.MinHeight = 24f;
-                    customInspector.BuildInspectorUI(ui);
+                    afterHeaderBranchLabel = (Label)instruction.operand;
+                    branchFound = true;
                 }
-                catch (Exception ex)
+                if (!branchFound && afterHeaderBranchLabel != null && !headerDone)
                 {
-                    ui.Text((LocaleString)"EXCEPTION BUILDING UI. See log");
-                    UniLog.Error(ex.ToString(), stackTrace: false);
+                    // check Enabled
+                    yield return new CodeInstruction(OpCodes.Call, _getEnabledMethod);
+                    yield return new CodeInstruction(OpCodes.Brfalse, afterHeaderPatchLabel);
+
+                    // do header patch
+                    yield return new CodeInstruction(OpCodes.Ldloc_0);
+                    yield return new CodeInstruction(OpCodes.Ldarg, 0);
+                    yield return new CodeInstruction(OpCodes.Ldarg, 1);
+                    yield return new CodeInstruction(OpCodes.Ldarg, 2);
+                    yield return new CodeInstruction(OpCodes.Ldarg, 3);
+                    yield return new CodeInstruction(OpCodes.Ldarg, 4);
+                    yield return new CodeInstruction(OpCodes.Ldarg, 5);
+                    yield return new CodeInstruction(OpCodes.Call, _buildHeaderMethod);
+
+                    // skip original if did patch
+                    yield return new CodeInstruction(OpCodes.Br, afterHeaderBranchLabel);
+
+                    // mark after patch (start of original)
+                    yield return new CodeInstruction(OpCodes.Nop) { labels = [afterHeaderPatchLabel] };
+                    headerDone = true;
                 }
-            }
-            else
-            {
-                WorkerInspector.BuildInspectorUI(worker, ui, memberFilter);
-            }
+                // this patch calls OnBuildInspectorHeaderText unconditionally (even if there is no InspectorHeaderAttribute)
+                if (afterHeaderTextBranchLabel is null && !headerTextDone &&
+                    instruction.opcode == OpCodes.Ldloc_1 && 
+                    instArr[i-1].opcode == OpCodes.Stloc_1 && 
+                    instArr[i-2].Calls(AccessTools.Method(typeof(CustomAttributeExtensions), nameof(CustomAttributeExtensions.GetCustomAttribute), [typeof(MemberInfo)], [typeof(InspectorHeaderAttribute)])))
```

Comment:
declare this and the other method infos as variables at the start of the transpiler - doing the reflection in the loop isn't ideal and we don't need to store the methodinfos for later either

Reply:
lines 120 and 125 too